simulator/func_sim/alu.h

```diff
@@ -348,51 +320,46 @@ struct ALU
         }
     }
 
-    template<typename I> static
-    void eret( I* instr)
+    static void eret( I* instr)
     {
         // FIXME(pikryukov): That should behave differently for ErrorEPC
         jump( instr, instr->v_src[0]);
         instr->v_dst[0] &= instr->v_src[1] & ~(1U << 2);
     }
 
     // Traps
-    template<typename I> static void breakpoint( I* instr)   { instr->trap = Trap::BREAKPOINT; }
-    template<typename I> static void syscall   ( I* instr)   { instr->trap = Trap::SYSCALL;    }
-    template<typename I> static void halt( I* instr)   { instr->trap = Trap::HALT; }
-    template<typename I, Predicate<I> p> static void trap( I* instr) { if (p( instr)) instr->trap = Trap::EXPLICIT_TRAP; }
-    template<typename I> static void unknown_instruction( I* instr) { instr->trap = Trap::UNKNOWN_INSTRUCTION; }
-
-    template<typename I> static
-    void csrrw( I* instr)
+    static void breakpoint( I* instr)   { instr->trap = Trap::BREAKPOINT; }
+    static void syscall   ( I* instr)   { instr->trap = Trap::SYSCALL;    }
+    static void halt( I* instr)   { instr->trap = Trap::HALT; }
+    template<Predicate p> static void trap( I* instr) { if (p( instr)) instr->trap = Trap::EXPLICIT_TRAP; }
+    static void unknown_instruction( I* instr) { instr->trap = Trap::UNKNOWN_INSTRUCTION; }
+
+    static void csrrw( I* instr)
     {
         instr->v_dst[0]  = instr->v_src[0]; // CSR <- RS1
         instr->v_dst[1] = instr->v_src[1]; // RD  <- CSR
     }
 
-    template<typename I> static
-    void csrrs( I* instr)
+    static void csrrs( I* instr)
     {
         instr->mask   = instr->v_src[0];
         instr->v_dst[0]  = all_ones<typename I::RegisterUInt>(); // CSR <- 0xffff & RS1
         instr->v_dst[1] = instr->v_src[1]; // RD <- CSR
     }
 
-    template<typename I> static
-    void csrrwi( I* instr)
+    static void csrrwi( I* instr)
     {
         instr->v_dst[0]  = instr->v_imm;  // CSR <- RS1
         instr->v_dst[1] = instr->v_src[1]; // RD  <- CSR
     }
 
-    template<typename I, typename T> static
+    template <typename T = RegUInt> static
```

Comment:
Please keep indentation consistent.

Reply:
I still see the extra space between `template` and `typename` in some definitions, which should not be here